### DISCUSSION TITLE
Requesting a Sample code to Scale object using Transformable Shape and Vertices Manipulation

### DISCUSSION THREAD

**Discussion by matangpanchal** [bepu/bepuphysics1#26] [Q&A]
Hello Sir,
In my project I wish to scale convex mesh if certain conditions are met, I gone through several threads in older bepu forums and and I found that everywhere you have suggested to bake vertices of the shape or use Transformable shape and apply linear transformation. 
I tried transformable shape method and applied required scale. Unfortunately the convex shape in Viewport doesn't show scaled , but it generates contacts as per the new scaled instance. As per my requirement the effect should be also be visible in viewport along with correct contact generation.  So can you please show a piece of code which accomplish this task using Vertices Bake method as well as Transformable shape method.

**Comment by RossNordby** [comment]
The physical and graphical representations are independent; if you want the graphical representation to change, you'll need to also apply the transform to it. That's not related to bepuphysics1, though; the implementation would depend on how you're handling graphics.

(The bepuphysics1 demos renderer has hardcoded awareness of the physical shape types, and knows how to make graphical meshes for them. I wouldn't recommend using the demos renderer for anything, or even really borrowing its architecture, but the TransformableShapes go through [DisplayConvex](https://github.com/bepu/bepuphysics1/blob/60c236903a39f1542e60109f0e90e8f4b27ac652/BEPUphysicsDrawer/Models/Display%20types/Entity%20types/DisplayConvex.cs). It samples the shape in a bunch of directions to create a mesh from it. Again, this is very likely not what you would want to do in your own renderer; just applying the transform to the graphical model when rendering would be easier and faster.)
